bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSslUtil.java

```diff
@@ -0,0 +1,185 @@
+            File file = new File(keystorePath);
+            if (!file.exists()) {
+                // create new keystore
+                logger.info("Creating new keystore {} because it doesn't exist.", keystorePath);
+                return createKeyStore(keystorePath, keystorePassword);
+            } else {
+                // load keystore as a first check
+                KeyStore keyStore = KeyStore.getInstance("JKS");
+                keyStore.load(new FileInputStream(file), keystorePassword.toCharArray());
+                logger.debug("Using existing keystore {}", keystorePath);
+                return keyStore;
+            }
+        } catch (OperatorCreationException | GeneralSecurityException | IOException e) {
+            logger.debug("Exception during keystore creation {}", e.getMessage());
+            throw new PairingFailedException("Can not create or load keystore file: " + keystorePath
+                    + ". Check path, write access and JKS content.", e);
+        }
+    }
+
+    private X509Certificate generateClientCertificate(KeyPair keyPair)
+            throws GeneralSecurityException, IOException, OperatorCreationException {
+        final String dirName = "CN=" + getBoschSHCId() + ", O=openHAB, L=None, ST=None, C=None";
+        logger.debug("Creating a new self signed certificate: {}", dirName);
+        final Instant now = Instant.now();
+        final Date notBefore = Date.from(now);
+        final Date notAfter = Date.from(now.plus(Duration.ofDays(365 * 10)));
+        X500Name name = new X500Name(dirName);
+
+        // create the certificate
+        X509v3CertificateBuilder certificateBuilder = new JcaX509v3CertificateBuilder(name, // Issuer
+                BigInteger.valueOf(now.toEpochMilli()), notBefore, notAfter, name, // Subject
+                keyPair.getPublic() // Public key to be associated with the certificate
+        );
+        // and sign it
+        ContentSigner contentSigner = new JcaContentSignerBuilder("SHA256WithRSA").build(keyPair.getPrivate());
+        return new JcaX509CertificateConverter().setProvider(new BouncyCastleProvider())
+                .getCertificate(certificateBuilder.build(contentSigner));
+    }
+
+    private KeyStore createKeyStore(String keystore, String keystorePassword)
+            throws IOException, OperatorCreationException, GeneralSecurityException {
+        // create a new keystore
+        KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(null, null);
+
+        // create new key pair for BoschSHC binding
+        logger.debug("Creating new keypair");
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+        kpg.initialize(2048);
+        KeyPair keyPair = kpg.generateKeyPair();
+
+        Security.addProvider(new BouncyCastleProvider());
+        Signature signer = Signature.getInstance("SHA256withRSA", "BC");
+        signer.initSign(keyPair.getPrivate());
+        signer.update("Hello OpenHAB".getBytes());
```

Comment:
The default charset varies from system to system so it is better to specify it yourself instead of using the default.

Reply:
Using StandardCharsets.UTF_8